spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -88,22 +90,16 @@ public URI getURI() throws IOException {
 	}
 
 	private Blob resolve() throws IOException {
-		synchronized (this.monitor) {
-			if (this.blob == null) {
-				try {
-					URI uri = getURI();
-					BlobId blobId = BlobId.of(uri.getHost(),
-							uri.getPath().substring(1, uri.getPath().length()));
-					this.blob = this.storage.get(blobId);
-					return this.blob;
-				}
-				catch (Exception e) {
-					throw new IOException(
-							"Failed to open remote connection to " + this.location, e);
-				}
-			}
+		try {
+			URI uri = getURI();
+			BlobId blobId = BlobId.of(uri.getHost(),
+					uri.getPath().substring(1, uri.getPath().length()));
+			return this.storage.get(blobId);
+		}
+		catch (Exception e) {
```

Comment:
This catch is a bit strange. Only getURI() throws an `IOException`, yet we're catching every `Exception`.

Do we really need that extra error message? If so, I'd suggest we only catch `IOException`, instead. Otherwise, get rid of the catch altogether.